### PR TITLE
Fix incorrect "avoid" set in globenv extra data

### LINKED ISSUE #13348: Regression: TC resolution failure with name mangling
#### Description of the problem

[std++'s `base.v`](https://gitlab.mpi-sws.org/iris/stdpp/-/blob/master/theories/base.v) fails to build with name mangling on latest Coq master. It used to work until a few days ago.

Lemma that fails:
```coq
Instance impl_inhabited {A} `{Inhabited B} : Inhabited (A → B) :=
  populate (λ _, inhabitant).
```
Error message:
```
Error:
Unable to satisfy the following constraints:
In environment:
A : Type
B : Type
_3 : Inhabited B
_3 : A

?_0 : "Inhabited B"
```

Last known-to-work commit: dfdecf24
First known-to-be-broken commit: fa6c67d7
Commit range: https://github.com/coq/coq/compare/dfdecf2...fa6c67d

### PR DESCRIPTION
**Pull request by SkySkimmer**
Fix #13348

### CODE REVIEW

File: engine/evarutil.ml [hunk 1]
```diff
@@ -371,7 +371,7 @@ let push_rel_decl_to_named_context
       let subst = update_var id0 id subst in
       let d = decl |> NamedDecl.of_rel_decl (fun _ -> id0) |> map_decl (csubst_subst subst) in
       let nc = replace_var_named_declaration id0 id nc in
-      (push_var id0 subst, Id.Set.add id avoid, push_named_context_val d nc)
+      (push_var id0 subst, Id.Set.add id0 avoid, push_named_context_val d nc)
```
Comment (SkySkimmer) [inline engine/evarutil.ml:374]: Not sure if this is correct or if we should add both id and id0.

Comment (SkySkimmer) [reply engine/evarutil.ml]: CI says we need both https://gitlab.com/coq/coq/-/jobs/843553455

**Review by ppedrot — APPROVED** [review]
I am a bit surprised that this doesn't create backwards incompatibility, but CI seems to be fine so that's good enough for me.

**Comment by herbelin** [comment]
Intuitively, it is indeed being lucky that there are no more incompatibilities.

**Comment by ppedrot** [comment]
I will wait for Iris to be fixed before merging this PR, just in case.

**Comment by Zimmi48** [comment]
OK, I'm going to backport the last batch of PR, so I'll postpone this one to 8.13 then.

**Comment by ppedrot** [comment]
@SkySkimmer can you rebase for a clean CI?

**Pull request merged**